diff -ur libtasn1-4.1-orig/lib/decoding.c libtasn1-4.1/lib/decoding.c
--- libtasn1-4.1-orig/lib/decoding.c	2014-08-20 21:36:55.000000000 +0200
+++ libtasn1-4.1/lib/decoding.c	2014-09-04 12:41:11.504502584 +0200
@@ -731,6 +731,7 @@
 {
   int len2, len3, counter, tot_len, indefinite;
   int result;
+  int orig_der_len = der_len;
 
   counter = 0;
 
@@ -792,7 +793,7 @@
 	  asn1_length_der (tot_len, temp, &len2);
 	  _asn1_set_value (node, temp, len2);
 
-	  ret = _asn1_extract_der_octet (node, der, der_len, flags);
+	  ret = _asn1_extract_der_octet (node, der, orig_der_len, flags);
 	  if (ret != ASN1_SUCCESS)
 	    return ret;
 
