diff -ur libtasn1-3.8.orig/lib/decoding.c libtasn1-3.8/lib/decoding.c
--- libtasn1-3.8.orig/lib/decoding.c	2014-08-23 09:26:09.000000000 +0200
+++ libtasn1-3.8/lib/decoding.c	2014-09-04 10:38:51.964592251 +0200
@@ -720,6 +720,7 @@
 {
   int len2, len3, counter, tot_len, indefinite;
   int result;
+  int orig_der_len = der_len;
 
   counter = 0;
 
@@ -777,7 +778,7 @@
 	  asn1_length_der (tot_len, temp, &len2);
 	  _asn1_set_value (node, temp, len2);
 
-	  ret = _asn1_extract_der_octet (node, der, der_len);
+	  ret = _asn1_extract_der_octet (node, der, orig_der_len);
 	  if (ret != ASN1_SUCCESS)
 	    return ret;
 
